week1/plot_trips.R

```diff
@@ -1,62 +1,186 @@
+trips %>%
+  select(birth_year, gender) %>%
+  mutate(age = 2014 - birth_year) %>%
+  count(age, gender) %>%
+  filter(!(n > 1250 & gender == 'Unknown')) %>%
+  ggplot(aes(x = age, y = n)) +
+  geom_line(aes(color = gender)) +
+  geom_point(aes(color = gender)) + 
+  scale_y_continuous(label=comma) + 
+  facet_wrap(~ gender, ncol = 1, scales = 'free') 
+  
+
+# plot the ratio of male to female trips (on the y axis) by age (on the x axis)
+# hint: use the spread() function to reshape things to make it easier to compute this ratio
+# (you can skip this and come back to it tomorrow if we haven't covered spread() yet)
+trips %>%
+  select(gender, birth_year) %>%
+  mutate(age = 2014 - birth_year) %>%
+  group_by(age, gender) %>% 
+  summarise(count = n()) %>%
+  pivot_wider(names_from = gender, values_from = count) %>% 
+  ggplot(aes(x = age, y = Male/Female)) +
+  geom_point() +
+  geom_line() +
+  geom_smooth()
+  
+
+
+
+########################################
+# plot weather data
+########################################
+# plot the minimum temperature (on the y axis) over each day (on the x axis)
+weather %>%
+  ggplot(aes(x=ymd, y = tmin)) +
+  geom_point(aes(color = factor(month(ymd), levels= 1:12, labels = month.name))) + 
+  geom_smooth() +
+  labs(x = 'days', y = 'min_temperature', color = 'Month')
+
+
+
+# plot the minimum temperature and maximum temperature (on the y axis, with different colors) over each day (on the x axis)
+# hint: try using the gather() function for this to reshape things before plotting
+# (you can skip this and come back to it tomorrow if we haven't covered reshaping data yet)
+weather %>%
+  select(ymd, tmax, tmin) %>%
+  pivot_longer(cols = c("tmax", "tmin"), names_to = "temperature_type", values_to = "temperature") %>%
+  ggplot(aes(x = ymd, y = temperature, color = temperature_type)) +
+  geom_point() +
+  geom_smooth()
+
+
+
+########################################
+# plot trip and weather data
+########################################
+
+# join trips and weather
+trips_with_weather <- inner_join(trips, weather, by="ymd")
+
+# plot the number of trips as a function of the minimum temperature, where each point represents a day
+# you'll need to summarize the trips and join to the weather data to do this
+trips_with_weather %>%
+  select(tmin, ymd) %>%
+  group_by(ymd, tmin) %>%
+  summarize(count = n()) %>%
+  ggplot(aes(x = tmin, y = count)) + 
+  geom_point(aes(size = count, color = tmin))
+
+
+# repeat this, splitting results by whether there was substantial precipitation or not
+# you'll need to decide what constitutes "substantial precipitation" and create a new T/F column to indicate this
+trips_with_weather %>%
+  select(tmin, ymd, prcp) %>%
+  mutate(prcp_test = prcp > 0.1) %>% 
+  group_by(ymd, tmin, prcp_test) %>%
+  summarise(count = n()) %>%
+  ungroup() %>%
+  ggplot(aes(x = tmin, y = count, color = prcp_test)) + 
+  geom_point() +
+  facet_wrap(~ prcp_test) +

```

Comment:
you can also assign graphs to variables - and refernece them later with addition.